src/odb/include/odb/db.h

```diff
@@ -7982,8 +7998,9 @@ class dbModInst : public dbObject
   static dbModInst* getModInst(dbBlock* block_, uint dbid_);
 
   /// Swap the module of this instance.
-  /// Returns true if the operations succeeds.
-  bool swapMaster(dbModule* module);
+  /// Returns new mod inst if the operations succeeds.
+  /// Old mod inst is deleted along with its child insts.
+  dbModInst* swapMaster(dbModule* module);
```

Comment:
Why delete the dbModInst rather than bind it to the new master (like dbInst::swapMaster)?

Reply:
The high level operation is just composed of atomic operations.  Why can't any "pointer swap" be recorded?